### DISCUSSION TITLE
LLVM's opt Absence in Emscripten's Upstream

### DISCUSSION THREAD

**Discussion by merlinsun** [emscripten-core/emscripten#21459] [General]
Hello,

I've observed that LLVM's opt is not present in the emsdk's upstream path. I was wondering if there have been any changes to Emscripten's use of opt for optimization.

Thank you in advance for your assistance. :)

**Comment by sbc100** [comment]
Emscripten doesn't use `opt` directly for anything anymore.    All llvm passes are now run via clang, just like they would be for normal clang users.

**Reply by merlinsun** [reply]
Got it.
Thank you very much!
